### PR TITLE
Making into Python library

### PR DESCRIPTION
**Pull request by stefanwebb**
I've made two suggested changes here.

1. The repo becomes a library that you can install with, e.g., `python setup.py develop`.
2. The models are loaded by calling `import pyro_models`, `models = pyro_models.load()`.

`models` is a dictionary with metadata on all the models including all the `model` stochastic functions. The data is loaded with a helper function from this metadata.

### CODE REVIEW

File: pyro_models/load.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+            # NOTE: Need to use partial to make sure correct foo is bound to function!
+            model['model'] = partial(model_wrapped, foo)
+
+        elif filename.endswith('.py.json'):
+            model['data_file'] = os.path.join(model_dir, f)
+        else:
+            raise Exception(f'Invalid file {f} in model zoo')
+
+    # TODO: Check that there is a data file for every model file
```
Comment (jpchen) [inline pyro_models/load.py]: this has already been done in a preprocessing step. those that didnt were omitted.

File: pyro_models/load.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+        if filename.endswith('.py'):
+            model['model_file'] = os.path.join(model_dir, f)
+            model['model_name'] = filename[:-3].replace('_', ' ').lower()
+            model['index'] = number
+            foo = imp.load_source('model.'+filename[:-3].lower(), os.path.join(model_dir, f))
+            model['module'] = foo 
+
+            # NOTE: Need to use partial to make sure correct foo is bound to function!
+            model['model'] = partial(model_wrapped, foo)
```
Comment (jpchen) [inline pyro_models/load.py:29]: 👍

File: pyro_models/utils.py [hunk 3]
```diff
@@ -51,34 +44,17 @@ def tensorize_data(data):
...
     # convert dicts to torch tensors
     tensorize_data(data)
-    if hasattr(module, 'transformed_data'):
-        # run transformed_data block if it exists
-        module.transformed_data(data)
-    for i in range(args.num_epochs):
-        params = {}
-        loss = svi.step(data, params)
-        print(loss)
+    #print('tensorized data', data)
```
Comment (jpchen) [inline pyro_models/utils.py]: can you remove this debugging prints?

**Review by jpchen — APPROVED** [review]
LGTM, thanks for the changes! if you could edit the readme with the run command and the description above of how to import it as a module, that'd be great (or in a follow up PR)

**Comment by stefanwebb** [comment]
No worries, I've made all those changes :)

**Pull request merged**